src/main/java/uk/gov/hmcts/reform/sscs/service/NotificationService.java

```diff
@@ -60,51 +67,91 @@ private void sendNotificationPerSubscription(NotificationWrapper notificationWra
         for (SubscriptionWithType subscriptionWithType : notificationWrapper.getSubscriptionsBasedOnNotificationType()) {
 
             if (isValidNotification(notificationWrapper, subscriptionWithType.getSubscription(), notificationType)) {
-
-                Notification notification = notificationFactory.create(notificationWrapper, subscriptionWithType.getSubscriptionType());
-
-                sendNotificationService.sendEmailSmsLetterNotification(notificationWrapper, subscriptionWithType.getSubscription(), notification);
-                processOldSubscriptionNotifications(notificationWrapper, notification);
+                sendNotification(notificationWrapper, subscriptionWithType);
                 reminderService.createReminders(notificationWrapper);
+                resendLastNotification(notificationWrapper, subscriptionWithType);
             }
         }
     }
 
+    private void resendLastNotification(NotificationWrapper notificationWrapper, SubscriptionWithType subscriptionWithType) {
+        if (shouldProcessLastNotification(notificationWrapper, subscriptionWithType)) {
+            log.info("resending the last notification");
```

Comment:
Can we add some more details to this log message so we know what notification is resent and which case id?